surface/include/pcl/surface/advancing_front_utils.h

```diff
@@ -0,0 +1,310 @@
+#ifndef PCL_SURFACE_ADVANCING_FRONT_UTILS_H_
+#define PCL_SURFACE_ADVANCING_FRONT_UTILS_H_
+
+#include <pcl/surface/advancing_front_point_type.h>
+#include <pcl/point_types.h>
+
+namespace pcl
+{
+  namespace afront
+  {
+
+    /** \brief Align pn's normal with av so they point in the same direction */
+    bool
+    alignNormal (Eigen::Vector3f &pn, const Eigen::Vector3f &av)
+    {
+      double dot = pn.dot (av);
+      if (dot < 0.0)
+      {
+        pn *= -1.0;
+        return true;
+      }
+
+      return false;
+    }
+
+    /** \brief Align pn's normal with av so they point in the same direction */
+    bool
+    alignNormal (AfrontVertexPointType &pn, const AfrontVertexPointType &av)
+    {
+      Eigen::Vector3f normal = pn.getNormalVector3fMap ();
+      if (alignNormal (normal, av.getNormalVector3fMap ()))
+      {
+        pn.normal_x = normal[0];
+        pn.normal_y = normal[1];
+        pn.normal_z = normal[2];
+        return true;
+      }
+
+      return false;
+    }
+
+    /** \brief Check if two normals are within a tolerance */
+    bool
+    checkNormal (const Eigen::Vector3f n1, const Eigen::Vector3f n2, const double &tol)
+    {
+      double dot = n1.dot (n2);
+      if (dot < 0.0)
+        return false;
+
+      double denom = n1.norm () * n2.norm ();
+      double angle = acos (dot / denom);
```

Comment:
Please use `std::acos`. Also, this function allows denormalized normals. What if one of the norms is zero?